presto-main/src/main/java/io/prestosql/spiller/FileSingleStreamSpillerFactory.java

```diff
@@ -215,11 +216,43 @@ private synchronized Path getNextSpillPath()
     private boolean hasEnoughDiskSpace(Path path)
     {
         try {
-            FileStore fileStore = getFileStore(path);
-            return fileStore.getUsableSpace() > fileStore.getTotalSpace() * (1.0 - maxUsedSpaceThreshold);
+            HetuFileSystemClient fileSystemClient = getFileSystem(path, spillToHdfs, spillProfile, fileSystemClientManager);
+            return fileSystemClient.getUsableSpace(path) > fileSystemClient.getTotalSpace(path) * (1.0 - maxUsedSpaceThreshold);
         }
         catch (IOException e) {
             throw new PrestoException(OUT_OF_SPILL_SPACE, "Cannot determine free space for spill", e);
         }
     }
+
+    private void createSpillDirectories()
+    {
+        if (!spillDirectoriesCreated) {
```

Comment:
May be random number alone is not sufficient; either use a UUID or prepend WorkerID/Name for keeping it unique or create subfolder by worker name in the HDFS spill path